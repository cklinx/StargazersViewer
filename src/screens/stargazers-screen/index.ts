export {StargazersScreen} from './stargazers-screen';
