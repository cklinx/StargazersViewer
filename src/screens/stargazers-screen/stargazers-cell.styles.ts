import {StyleSheet} from 'react-native';
import {Fonts} from '@stargazers/theme';

export const styles = StyleSheet.create({
  cellWrapper: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'space-between',
  },
  cell: {
    alignItems: 'flex-start',
    borderBottomColor: '#F4f5f7',
    borderBottomWidth: 1,
    padding: 8,
    paddingVertical: 10,
    justifyContent: 'center',
    minHeight: 90,
  },
  imageWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
    marginEnd: 20,
  },
  badge: {
    width: 20,
    height: 20,
    backgroundColor: '#FFF',
    borderWidth: 0,
    borderRadius: 3,
    overflow: 'hidden',
    fontSize: 12,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#FFF',
    position: 'absolute',
    paddingTop: 2,
    textAlignVertical: 'center',
    right: 5,
    top: 10,
  },
  productImage: {
    width: 90,
    height: 90,
    position: 'relative',
  },
  placeholder: {
    resizeMode: 'cover',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%',
  },
  textsContainer: {
    fontFamily: Fonts.regular,
    overflow: 'hidden',
    flex: 1,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    textShadowOffset: {width: 0, height: 0},
  },
  rightColumn: {
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    alignContent: 'center',
    flexDirection: 'column',
    right: 3,
    marginTop: 0
  },
  productName: {
    fontFamily: Fonts.helveticaMedium,
    fontSize: 18,
    lineHeight: 18,
    color: '#281d6c',
    marginBottom: 2,
    paddingTop: 10,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0,
    shadowRadius: 0,
  },
  updatedAtWrapper: {
    flexDirection: 'row',
    display: 'flex',
    justifyContent: 'flex-start',
    marginTop: 4,
    height: 18,
  },
  updatedAt: {
    fontSize: 12,
    color: '#281d6c',
    marginLeft: 0,
    lineHeight: 18,
    fontFamily: Fonts.regular,
    flex: 1,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0,
    shadowRadius: 0,
    paddingBottom: 5,
  },
  counterRowsWrapper: {
    flexDirection: 'row',
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    alignContent: 'center',
    marginTop: 6,
  },
  starCounterImage: {
    width: 12,
    height: 12,
    alignSelf: 'center',
    alignItems: 'center',
    marginEnd: 5,
  },
  starCounterText: {
    marginLeft: 0,
    fontSize: 12,
    fontFamily: Fonts.regular,
  },
});
