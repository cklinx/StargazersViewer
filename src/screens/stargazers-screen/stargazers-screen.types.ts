import {ListRenderItemInfo} from 'react-native';
import {StargazersItem} from '@stargazers/services/user-service/user-service.types';

export interface StargazersCellProps extends ListRenderItemInfo<StargazersItem> {
  onStargazerClick: (item: StargazersItem) => void;
}
