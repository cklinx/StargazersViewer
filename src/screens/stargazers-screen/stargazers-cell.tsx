import React from 'react';
import {Image, Text, TouchableHighlight, View} from 'react-native';
import {Colors} from '@stargazers/theme';
import {styles} from '@stargazers/screens/repositories-screen/repository-cell.styles';
import useColorScheme from '@stargazers/hooks/useColorScheme';
import {StargazersCellProps} from '@stargazers/screens/stargazers-screen/stargazers-screen.types';
import Icon from 'react-native-vector-icons/Feather';
import Formatter from '@stargazers/utils/formatter';

export const StargazersCell: React.FC<StargazersCellProps> = ({
  item,
  separators,
  onStargazerClick,
}) => {
  const colorScheme = useColorScheme();

  return (
    <>
      <TouchableHighlight
        style={{
          ...styles.cell,
          borderBottomColor: Colors[colorScheme].cellBorderBottomColor,
          backgroundColor: Colors[colorScheme].background,
        }}
        underlayColor={Colors[colorScheme].cellHighlight}
        onShowUnderlay={separators.highlight}
        onHideUnderlay={separators.unhighlight}
        onPress={() => onStargazerClick(item)}>
        <View style={styles.cellWrapper}>
          <View
            style={{
              ...styles.avatarWrapper,
              backgroundColor: Colors[colorScheme].background,
              borderColor: Colors[colorScheme].secondary,
            }}>
            <Image
              source={{
                uri: item.avatar_url,
              }}
              resizeMode={'cover'}
              style={styles.avatarImage}
            />
          </View>
          <View style={styles.textsContainer}>
            <Text
              style={{
                ...styles.productName,
                color: Colors[colorScheme].tint,
              }}
              numberOfLines={1}
              ellipsizeMode="tail">
              {item.login}
            </Text>

            <View style={styles.updatedAtWrapper}>
              <Text
                style={{
                  ...styles.updatedAt,
                  color: Colors[colorScheme].tint,
                }}>
                {Formatter.getStargazerPublicURL(item.url)}
              </Text>
            </View>
          </View>
          <View style={styles.rightColumn}>
            <Icon name={'link-2'} size={20} color={Colors[colorScheme].tint} />
          </View>
        </View>
      </TouchableHighlight>
    </>
  );
};
