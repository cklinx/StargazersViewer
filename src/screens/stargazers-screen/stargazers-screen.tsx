import React, {useEffect, useState} from 'react';
import useColorScheme from '../../hooks/useColorScheme';
import {FlatList, View, Text, Platform, RefreshControl, TextInput, Linking} from 'react-native';
import {getStargazers} from '@stargazers/services/user-service/user-service';
import {StargazersItem} from '@stargazers/services/user-service/user-service.types';
import {styles} from './stargazers-screen.styles';
import {StargazersScreenProps} from '@stargazers/navigators/navigator.types';
import {Colors, GlobalStyles} from '@stargazers/theme';
import {StargazersCell} from '@stargazers/screens/stargazers-screen/stargazers-cell';
import Formatter from '@stargazers/utils/formatter';

export const StargazersScreen = ({route, navigation}: StargazersScreenProps) => {
  const colorScheme = useColorScheme();
  const [refreshing, setRefreshing] = useState(false);
  const [loading, setLoading] = useState(false);
  const owner = route.params.owner;
  const repository = route.params.repository;

  const [stargazers, setStargazers] = useState<StargazersItem[]>(route.params.stargazers);
  const [filteredStargazers, setFilteredStargazers] = useState<StargazersItem[]>(
    route.params.stargazers,
  );

  useEffect(() => {
    navigation.setOptions({
      title: repository.name,
      headerBackTitle: 'Repos',
    });
  }, [repository]);

  const onRefresh = React.useCallback(() => {
    setRefreshing(true);
    getStargazers(owner, repository.name).then(
      response => {
        setStargazers(response.data);
        setRefreshing(false);
      },
      () => {
        setRefreshing(false);
      },
    );
  }, []);

  const searchFilterFunction = (text: string) => {
    if (text) {
      const newData = stargazers.filter(function (item) {
        const itemData = item.login ? item.login.toUpperCase() : ''.toUpperCase();
        const textData = text.toUpperCase();
        return itemData.indexOf(textData) > -1;
      });
      setFilteredStargazers(newData);
    } else {
      setFilteredStargazers(stargazers);
    }
  };

  const onStargazerClick = async (stargazer: StargazersItem) => {
    await Linking.openURL(Formatter.getStargazerPublicURL(stargazer.url));
  };

  // @ts-ignore
  return (
    <View
      style={{
        backgroundColor: Colors[colorScheme].background,
        flex: 1,
      }}>
      <View
        style={{
          ...styles.searchBarWrapper,
          backgroundColor: Colors[colorScheme].background,
        }}>
        <TextInput
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={searchFilterFunction}
          placeholder="Search stargazer.."
          placeholderTextColor={Colors[colorScheme].infoTextColor}
          style={{
            ...styles.searchBarText,
            color: Colors[colorScheme].infoTextColor,
            backgroundColor: Colors[colorScheme].secondary,
          }}
        />
      </View>
      <FlatList
        //@ts-ignore
        ItemSeparatorComponent={
          Platform.OS !== 'android' &&
          (({highlighted}) => <View style={[styles.separator, highlighted && {marginLeft: 0}]} />)
        }
        removeClippedSubviews={true}
        style={styles.scrollView}
        data={filteredStargazers}
        keyExtractor={item => item.id.toString()}
        renderItem={props => {
          return <StargazersCell {...props} onStargazerClick={onStargazerClick} />;
        }}
        refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh} />}
        ListHeaderComponent={() => {
          return (
            <View
              style={{
                ...styles.productCounterBar,
                backgroundColor: Colors[colorScheme].background,
              }}>
              <Text
                style={{
                  ...styles.productCountLabel,
                  color: Colors[colorScheme].secondary,
                }}>
                {filteredStargazers.length === 1
                  ? filteredStargazers.length + ' stargazer'
                  : filteredStargazers.length + ' stargazers'}
              </Text>
            </View>
          );
        }}
        stickyHeaderIndices={[0]}
        ListEmptyComponent={() => {
          return (
            <View style={GlobalStyles.emptyListWrapper}>
              <Text
                style={{
                  ...GlobalStyles.emptyListText,
                  color: Colors[colorScheme].tint,
                }}>
                no stargazer...yet
              </Text>
            </View>
          );
        }}
      />
    </View>
  );
};
