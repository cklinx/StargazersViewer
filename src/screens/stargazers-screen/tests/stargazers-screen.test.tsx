import React from 'react';

import renderer from 'react-test-renderer';
import {StargazersScreen} from '../stargazers-screen';
import {render} from '@testing-library/react-native';
import {FlatList} from 'react-native';
import {StargazersCell} from '@stargazers/screens/stargazers-screen/stargazers-cell';

jest.mock('react-native-vector-icons/Feather', () => 'Feather');
const goBack = jest.fn();
const navigate = jest.fn();
const setOptions = jest.fn();
const addListener = jest.fn();
type AppStarGazer = {
  id: number;
  avatarImage: string;
  login: string;
  url: string;
};
const mockStargazers: AppStarGazer[] = [
  {
    id: 1,
    avatarImage: 'test',
    login: 'test',
    url: 'test',
  },
  {
    id: 2,
    avatarImage: 'test',
    login: 'test',
    url: 'test',
  },
  {
    id: 3,
    avatarImage: 'test',
    login: 'test',
    url: 'test',
  },
];
const mockRoute = {
  params: {
    owner: 'test',
    repository: {name: 'test'},
    stargazers: mockStargazers,
  },
};
const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
      setOptions: jest.fn(),
    }),
  };
});
describe('StargazersScreen', () => {
  it('should render correctly', () => {
    const {toJSON} = renderer.create(
      <StargazersScreen
        //@ts-ignore
        navigation={{goBack, navigate, setOptions, addListener}}
        //@ts-ignore
        route={mockRoute}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render exact cell as per source', () => {
    const componentTree = render(
      <StargazersScreen
        //@ts-ignore
        navigation={{goBack, navigate, setOptions, addListener}}
        //@ts-ignore
        route={mockRoute}
      />,
    );

    expect(componentTree.UNSAFE_getAllByType(FlatList).length).toBe(1);
    expect(componentTree.UNSAFE_getAllByType(StargazersCell).length).toBe(mockStargazers.length);
  });
});
