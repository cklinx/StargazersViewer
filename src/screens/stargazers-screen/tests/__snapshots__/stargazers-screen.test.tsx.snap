// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StargazersScreen should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 4,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Search stargazer.."
      placeholderTextColor="#FFFFFF"
      style={
        Object {
          "backgroundColor": "#414757",
          "borderColor": "#333",
          "borderRadius": 25,
          "color": "#FFFFFF",
          "fontSize": 14,
          "height": 40,
          "paddingHorizontal": 15,
          "width": "100%",
        }
      }
    />
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "avatarImage": "test",
          "id": 1,
          "login": "test",
          "url": "test",
        },
        Object {
          "avatarImage": "test",
          "id": 2,
          "login": "test",
          "url": "test",
        },
        Object {
          "avatarImage": "test",
          "id": 3,
          "login": "test",
          "url": "test",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    style={Object {}}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "paddingHorizontal": 14,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#414757",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": 15,
                "padding": 3,
              }
            }
          >
            3 stargazers
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#fff",
              "borderBottomColor": "#f4f4f4",
              "borderBottomWidth": 1,
              "justifyContent": "center",
              "minHeight": 60,
              "padding": 8,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderColor": "#414757",
                  "borderRadius": 30,
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginEnd": 20,
                  "overflow": "hidden",
                  "width": 60,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 60,
                    "width": 60,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "fontFamily": "Montserrat-Regular",
                  "overflow": "hidden",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "textShadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "color": "#383B42",
                    "fontFamily": "HelveticaNeue-Medium",
                    "fontSize": 18,
                    "lineHeight": 18,
                    "marginBottom": 2,
                    "paddingTop": 8,
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                test
              </Text>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "height": 18,
                    "justifyContent": "flex-start",
                    "marginTop": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "flex": 1,
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "lineHeight": 18,
                      "marginLeft": 0,
                      "paddingBottom": 5,
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  test
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginTop": 0,
                  "right": 3,
                }
              }
            >
              <Feather
                color="#383B42"
                name="link-2"
                size={20}
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              false,
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#fff",
              "borderBottomColor": "#f4f4f4",
              "borderBottomWidth": 1,
              "justifyContent": "center",
              "minHeight": 60,
              "padding": 8,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderColor": "#414757",
                  "borderRadius": 30,
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginEnd": 20,
                  "overflow": "hidden",
                  "width": 60,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 60,
                    "width": 60,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "fontFamily": "Montserrat-Regular",
                  "overflow": "hidden",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "textShadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "color": "#383B42",
                    "fontFamily": "HelveticaNeue-Medium",
                    "fontSize": 18,
                    "lineHeight": 18,
                    "marginBottom": 2,
                    "paddingTop": 8,
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                test
              </Text>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "height": 18,
                    "justifyContent": "flex-start",
                    "marginTop": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "flex": 1,
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "lineHeight": 18,
                      "marginLeft": 0,
                      "paddingBottom": 5,
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  test
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginTop": 0,
                  "right": 3,
                }
              }
            >
              <Feather
                color="#383B42"
                name="link-2"
                size={20}
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              false,
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#fff",
              "borderBottomColor": "#f4f4f4",
              "borderBottomWidth": 1,
              "justifyContent": "center",
              "minHeight": 60,
              "padding": 8,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderColor": "#414757",
                  "borderRadius": 30,
                  "borderWidth": 2,
                  "height": 60,
                  "justifyContent": "center",
                  "marginEnd": 20,
                  "overflow": "hidden",
                  "width": 60,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 60,
                    "width": 60,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "fontFamily": "Montserrat-Regular",
                  "overflow": "hidden",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "textShadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                }
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "color": "#383B42",
                    "fontFamily": "HelveticaNeue-Medium",
                    "fontSize": 18,
                    "lineHeight": 18,
                    "marginBottom": 2,
                    "paddingTop": 8,
                    "shadowOffset": Object {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                test
              </Text>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "height": 18,
                    "justifyContent": "flex-start",
                    "marginTop": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#383B42",
                      "flex": 1,
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 12,
                      "lineHeight": 18,
                      "marginLeft": 0,
                      "paddingBottom": 5,
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  test
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "column",
                  "justifyContent": "center",
                  "marginTop": 0,
                  "right": 3,
                }
              }
            >
              <Feather
                color="#383B42"
                name="link-2"
                size={20}
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
